From 648d44f1707bba91baaf9fb28e625a904c696a47 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 17 Jan 2017 17:16:20 +0100
Subject: [PATCH 0939/1345] usb: gadget: u3d: keep the irq line masked until
 the threaded handler finished

commit  bba6c45860f73be6418e5e6910c7b6dd6fb7687a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Adding IRQF_ONESHOT flag allows to keep the irq line masked until the
threaded handler finshed. This prevents from multiple handling of the
same interrupt.

Change-Id: I9221a4d3ffda4f01d3198c588b1ac9b974a17ea4
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38537
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/gadget/udc/mvebu_u3d.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/udc/mvebu_u3d.c b/drivers/usb/gadget/udc/mvebu_u3d.c
index 73556d7..a3a47f6 100644
--- a/drivers/usb/gadget/udc/mvebu_u3d.c
+++ b/drivers/usb/gadget/udc/mvebu_u3d.c
@@ -2357,8 +2357,8 @@ static int mvc2_probe(struct platform_device *pdev)
 			ret = devm_request_any_context_irq(&pdev->dev,
 					       gpio_to_irq(cp->vbus_pin),
 					       mvc2_vbus_irq,
-					       IRQ_TYPE_EDGE_BOTH, "mvebu-u3d",
-					       cp);
+					       IRQ_TYPE_EDGE_BOTH | IRQF_ONESHOT,
+					       "mvebu-u3d", cp);
 			if (ret < 0) {
 				cp->vbus_pin = -ENODEV;
 				dev_warn(&pdev->dev,
-- 
1.7.9.5

